sanity-dataset-backup/action.yml

```
@@ -0,0 +1,48 @@
+name: Backup
+description: Backup Sanity dataset
+
+inputs:
+  sanity-read-token:
+    description: "Sanity token, used to export dataset"
+    required: true
+    default: ""
+  dataset-name:
+    description: "Name of dataset to back up"
+    required: false
+    default: "production"
+runs:
+  using: "composite"
+
+  steps:
+    - name: Determine retention period
+      shell: bash
+      run: |
+        if [ $(date '+%A') == 'Saturday' ]; then
+          if [ $(date '+%d') -le 8 ]; then
+            # 1st saturday of the month
+            echo "RETENTION_DAYS=90" >> $GITHUB_ENV
+          else
+            # Saturday but not 1st in the month
+            echo "RETENTION_DAYS=45" >> $GITHUB_ENV
+          fi
+        else
+          # Non-saturday
+          echo "RETENTION_DAYS=14" >> $GITHUB_ENV
+        fi
+    - name: Set backup file name
+      run: |
+        echo "BACKUP_FILE_NAME=backup-${{ inputs.dataset-name }}-$(date +%Y-%m-%d).tar.gz" >> $GITHUB_ENV
+    - name: Export dataset
+      shell: bash
+      working-directory: sanity
+      env:
+        SANITY_AUTH_TOKEN: ${{ inputs.sanity-read-token }}
+      run: npx sanity dataset export ${{ inputs.dataset-name }} backups/${{ env.BACKUP_FILE_NAME }}
+    - name: Upload backup.tar.gz and store for ${{ env.RETENTION_DAYS }} days
+      uses: actions/upload-artifact@v2
+      with:
+        name: backup-tarball
```

```suggestion
        name: ${{ env.BACKUP_FILE_NAME }}
```

(når jeg tenker meg om er det vel egentlig *bare* her vi trenger å ha et meningsbærende navn, men jaja :D)

---

Godt poeng. Men det skader ikke 🙂